pkg/core/proxy/integrations/mongo/operation.go

```diff
@@ -475,6 +477,128 @@ func extractSectionSingle(data string) (string, error) {
 	return content, nil
 }
 
+func processOpReply(expected, mongoRequest models.MongoRequest, replySpec *models.MongoOpReply, logger *zap.Logger) string {
```

Comment:
you can return boolean along with the string. If the reply is to be changed then bool will be true.